experimental/relativetime/src/provider.rs

```diff
@@ -106,19 +103,25 @@ pub struct PluralRulesCategoryMapping<'data> {
 )]
 pub struct SingularSubPattern<'data> {
     #[cfg_attr(feature = "serde", serde(borrow))]
-    /// The underlying pattern
+    /// The underlying pattern with the placeholder "{0}" removed.
     pub pattern: Cow<'data, str>,
-    /// Optional index which is absent if the pattern does not have substitution
-    pub index: Option<u8>,
+    /// Denotes the substitution position in the pattern.
+    /// Equals 255 if the pattern does not have a placeholder.
+    pub index: u8,
 }
 
 impl<'data> SingularSubPattern<'data> {
     /// Construct a singular sub pattern from a pattern
     pub fn try_from_str(value: &str) -> Result<Self, DataError> {
-        let index = value.find("{0}").map(|index| index as u8);
+        let index = value.find("{0}").unwrap_or(255);
```

Comment:
issue: we should just do `.find` here, and something like the following

```rust
let (index, pattern) = if let Some(index) = value.find(...) {
   if value >= 255 {
      return DataError::Custom("found placeholder too large to store")
   }
   (format ..., index)
} else {
   (value.to_string(), 255)
}

```

Reply:
Done.